avocado/core/job.py

```diff
@@ -488,6 +479,23 @@ def _run(self):
 
         return self.exitcode
 
+    def create_test_suite(self):
+        """
+        Creates the test suite for this Job
+
+        This is a public Job API as part of the documented Job phases
```

Comment:
This docstring is quite unusual. I'd define the public API in the documentation. If we want to be explicit on the code-level, I'd prefix the non-public API with some prefix (if not necessary from outside, `_` is an obvious marker, if necessary from outside, we could use `x_$method` to emphasize no one should use it unless they really know, what are they doing.